lib/insights/api/common/application_controller_mixins/parameters.rb

```diff
@@ -7,10 +7,25 @@ def self.included(other)
             other.include(OpenapiEnabled)
           end
 
-          def params_for_create
+          def params_for_create(opts = {})
             check_if_openapi_enabled
             # We already validate this with OpenAPI validator, that validates every request, so we shouldn't do it again here.
-            body_params.permit!
+            if opts[:writeable] == true
+              attr_list = api_doc_definition.all_attributes - api_doc_definition.read_only_attributes
+              body_params.permit(*request_body_schema_keys + attr_list)
+            else
+              body_params.permit!
+            end
+          end
+
+          def request_body_schema_keys
+            Insights::API::Common::OpenApi::Docs.instance[api_version].content.dig("components", "schemas", request_body_schema_name, "properties").keys
```

Comment:
@lindgrenj6 Is the because the required properties are on the body instead of in the object definitions. What would happen if it's defined in both the object and in body.
https://github.com/RedHatInsights/catalog-api/blob/10f2396699672a844c4064c089bd3b4e7f28b3d9/public/doc/openapi-3-v1.0.0.json#L129

Reply:
Nothing would change - this is looking up the schema reference found in the `requestBody` parameter. 

It takes the reference here:
https://github.com/RedHatInsights/catalog-api/blob/10f2396699672a844c4064c089bd3b4e7f28b3d9/public/doc/openapi-3-v1.0.0.json#L856-L865

And looks up the schema here:
https://github.com/RedHatInsights/catalog-api/blob/10f2396699672a844c4064c089bd3b4e7f28b3d9/public/doc/openapi-3-v1.0.0.json#L2843-L2853

Reply:
The openapi parser does the validation on the required keys - that isn't changing. This just checks the readOnly keys as well as looks up any extra ones specified in a `requestBody` schema.